### ISSUE TITLE
Error with paginated list of changes in the history endpoint

### ISSUE THREAD

**Issue by n1k0** [Kinto/kinto#1114]
Setup, latest kinto master:

In `kinto.ini`, set `kinto.paginate_by` to `10` and enable `kinto.plugins.history`.

Then:

```http
http -a test: PUT :8888/v1/buckets/foo
http -a test: PUT :8888/v1/buckets/foo/collections/bar
echo '{"data":{"n":1}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":2}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":3}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":4}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":5}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":6}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":7}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":8}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":9}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":10}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":11}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
echo '{"data":{"n":12}}' | http -a test: POST :8888/v1/buckets/foo/collections/bar/records
```

Take the latest `last_modified` of the created records (the one with `n: 12`, eg. `1488187867432`), and execute this query (the idea it to get more than `10` change entries):

```http
http -a test: ":8888/v1/buckets/foo/history?collection_id=bar&max_target.data.last_modified=1488187867432&_sort=-target.data.last_modified&resource_name=record"
```

We get a 500 error and a nice Python stack trace:

```python
ERROR:kinto.core.views.errors:"GET   /v1/buckets/foo/history?resource_name=record&collection_id=bar&max_target.data.last_modified=1488187867432&_sort=-target.data.last_modified" ? (? ms) 'target.data.last_modified' agent=HTTPie/0.9.2 authn_type=basicauth collection_id=history collection_timestamp=1488187867433 errno=None exception=Traceback (most recent call last):
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/tweens.py", line 22, in excview_tween
    response = handler(request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid_tm/__init__.py", line 119, in tm_tween
    reraise(*exc_info)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid_tm/compat.py", line 15, in reraise
    raise value
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid_tm/__init__.py", line 98, in tm_tween
    response = handler(request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/router.py", line 158, in handle_request
    view_name
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/view.py", line 547, in _call_view
    response = view_callable(context, request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/config/views.py", line 182, in __call__
    return view(context, request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/viewderivers.py", line 393, in attr_view
    return view(context, request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/viewderivers.py", line 371, in predicate_wrapper
    return view(context, request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/viewderivers.py", line 302, in _secured_view
    return view(context, request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/viewderivers.py", line 442, in rendered_view
    result = view(context, request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/pyramid/viewderivers.py", line 147, in _requestonly_view
    response = view(request)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/cornice/service.py", line 491, in wrapper
    response = view_()
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/kinto/core/resource/__init__.py", line 289, in collection_get
    next_page = self._next_page_url(sorting, limit, lastrecord, offset)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/kinto/core/resource/__init__.py", line 1060, in _next_page_url
    token = self._build_pagination_token(sorting, last_record, offset)
  File "/home/niko/work/kinto-http.js/.venv/lib/python3.5/site-packages/kinto/core/resource/__init__.py", line 1084, in _build_pagination_token
    token['last_record'][field] = last_record[field]
KeyError: 'target.data.last_modified' lang=None uid=35f724b781ae01543ed29fd50dc19ed3434594c9e97e40e97f136626c12a64ae
```

**Comment by n1k0** [comment]
From the error it seems we're not splitting dotted values to retrieve the matching nested dict, hence the uncaught key error.

**Comment by n1k0** [comment]
Would something be enough?

```diff
diff --git a/kinto/core/resource/__init__.py b/kinto/core/resource/__init__.py
index a4f9c89..237c6fc 100644
--- a/kinto/core/resource/__init__.py
+++ b/kinto/core/resource/__init__.py
@@ -1087,7 +1087,7 @@ class UserResource:
         }
 
         for field, _ in sorting:
-            token['last_record'][field] = last_record[field]
+            token['last_record'][field] = dict_subset(last_record, field)
 
         return encode64(json.dumps(token))
```

**Comment by n1k0** [comment]
Fix proposal sent in #1116.
